### PR TITLE
Update hibernate to allow for protection of eviction_disabled nodes

### PR DESCRIPTION
**Pull request by CastAIPhil**

### CODE REVIEW

File: app/cast_utils.py [hunk 1]
```diff
@@ -124,17 +124,24 @@ def create_hibernation_node(cluster_id: str, castai_api_token: str, instance_typ
...
 
-def delete_all_pausable_nodes(cluster_id: str, castai_api_token: str, hibernation_node_id: str, job_node_id=None):
+def delete_all_pausable_nodes(cluster_id: str, castai_api_token: str, hibernation_node_id: str, protect_eviction_disabled: bool, job_node_id=None):
     """" Delete all nodes through CAST AI mothership excluding hibernation node"""
     node_list_result = get_castai_nodes(cluster_id, castai_api_token)
-
+    logging.info(protect_eviction_disabled)
     for node in node_list_result["items"]:
+        skip=False
+        if "autoscaling.cast.ai/removal-disabled" in node["labels"] and protect_eviction_disabled=="true":
```
Comment (AugustinasS) [inline app/cast_utils.py]: if "autoscaling.cast.ai/removal-disabled" in node["labels"] and protect_eviction_disabled:

protect_eviction_disabled is bool meaning == True, not string "true".

File: app/cast_utils.py [hunk 2]
```diff
@@ -124,17 +124,24 @@ def create_hibernation_node(cluster_id: str, castai_api_token: str, instance_typ
         return nodeId
 
 
-def delete_all_pausable_nodes(cluster_id: str, castai_api_token: str, hibernation_node_id: str, job_node_id=None):
+def delete_all_pausable_nodes(cluster_id: str, castai_api_token: str, hibernation_node_id: str, protect_eviction_disabled: bool, job_node_id=None):
```
Comment (AugustinasS) [inline app/cast_utils.py]: I would recommend to use same words as in k8s label removal-disabled not eviction -> protect_removal_disabled

File: app/cast_utils.py [hunk 3]
```diff
@@ -124,17 +124,24 @@ def create_hibernation_node(cluster_id: str, castai_api_token: str, instance_typ
...
     node_list_result = get_castai_nodes(cluster_id, castai_api_token)
-
+    logging.info(protect_eviction_disabled)
     for node in node_list_result["items"]:
+        skip=False
+        if "autoscaling.cast.ai/removal-disabled" in node["labels"] and protect_eviction_disabled=="true":
+            if node["labels"]["autoscaling.cast.ai/removal-disabled"] == "true":
+                skip=True
         # drain/delete each node
         if node["id"] == hibernation_node_id or node["id"] == job_node_id:
```
Comment (AugustinasS) [inline app/cast_utils.py:136]: maybe it's worth putting this logic also under skip variable handling below

File: app/k8s_utils.py [hunk 4]
```diff
@@ -7,7 +7,7 @@ class K8sAPIError(Exception):
     pass
 
 
-def cordon_all_nodes(client, exclude_node_id: str):
+def cordon_all_nodes(client, protect_eviction_disabled, exclude_node_id: str):
```
Comment (AugustinasS) [inline app/k8s_utils.py]: protect_eviction_disabled: bool

File: app/k8s_utils.py [hunk 5]
```diff
@@ -20,7 +20,10 @@ def cordon_all_nodes(client, exclude_node_id: str):
     node_list = client.list_node()
     for node in node_list.items:
         logging.info("Inspecting node %s to cordon", node.metadata.name)
-        if node.metadata.labels.get("provisioner.cast.ai/node-id") != exclude_node_id:
+        skip=False
+        if node.metadata.labels.get("autoscaling.cast.ai/removal-disabled")=="true" and protect_eviction_disabled=="true":
```
Comment (AugustinasS) [inline app/k8s_utils.py]: same as before, protect_eviction_disabled is not a string

**Review by AugustinasS — CHANGES_REQUESTED** [review]

File: app/k8s_utils.py (hunk 5 above)
Comment (CastAIPhil) [reply app/k8s_utils.py]: Made eviction_disabled a string and renamed it "removal_disabled"

File: app/k8s_utils.py (hunk 4 above)
Comment (CastAIPhil) [reply app/k8s_utils.py]: Changed it to a string and updated name.

File: app/cast_utils.py (hunk 3 above)
Comment (CastAIPhil) [reply app/cast_utils.py:136]: Tried a couple ways and this still seemed the cleanest, putting it down below made it hard to read the code.

File: app/cast_utils.py (hunk 2 above)
Comment (CastAIPhil) [reply app/cast_utils.py]: Updated

File: app/cast_utils.py (hunk 1 above)
Comment (CastAIPhil) [reply app/cast_utils.py]: Updated

**Review by AugustinasS — APPROVED** [review]

**Pull request merged**